client/remoterepo.c

```diff
@@ -668,8 +668,12 @@ TDNFDownloadFile(
     }
     else
     {
-        dwError = rename(pszFileTmp, pszFile);
-        BAIL_ON_TDNF_ERROR(dwError);
+        if (rename(pszFileTmp, pszFile) == -1)
+        {
+            dwError = errno;
+            BAIL_ON_TDNF_SYSTEM_ERROR(dwError);
+        }
+
         dwError = chmod(pszFile, S_IRUSR|S_IWUSR|S_IRGRP|S_IROTH);
```

Comment:
we should also do the same for chmod. if we get ret val as -1, look at errno and set dwError accordingly and do BAIL_ON_TDNF_SYSTEM_ERROR

Reply:
How could I miss that? Thank you! Fixed.